docs/manual/config/keys.rst

```diff
@@ -45,6 +45,82 @@ The :class:`EzKey` modifier keys (i.e. ``MASC``) can be overwritten through the
        'C': 'control',
     }
 
+KeyChords
+=========
+
+Qtile also allows sequences of keys to trigger callbacks. In Qtile, these
+sequences are known as chords and are defined with
+:class:`libqtile.config.KeyChord`. Chords are added to the ``keys`` section of
+the config file.
+
+::
+
+    from libqtile.config import Key, KeyChord
+
+    keys = [
+        KeyChord([mod], "z", [
+            Key([], "x", lazy.spawn("xterm"))
+        ])
+    ]
+
+The above code will launch xterm when the user presses Mod + z, followed by x.
+
+.. warning::
+    Users should note that key chords are aborted by pressing <escape>. In the
+    above example, if the user presses Mod + z, any following key presses will
+    still be sent to the currently focussed window. If <escape> has not been
+    pressed, the next press of x will launch xterm.
+
+Modes
+-----
+
+Chords can optionally specify a "mode". When this is done, the mode will remain
+active until the user presses <escape>. This can be useful for configuring a
+subset of commands for a particular situations (i.e. similar to vim modes).
+
+::
+
+    from libqtile.config import Key, KeyChord
+
+    keys = [
+        KeyChord([mod], "z", [
+            Key([], "plus", lazy.layout.grow()),
+            Key([], "minus", lazy.layout.shrink()),
+            Key([], "n", lazy.layout.normalize()),
+            Key([], "m", lazy.layout.maximize())],
+            mode="Windows"
+        )
+    ]
+
+In the above example, pressing Mod + z triggers the "Windows" mode. Users can
+then resize windows by just pressing g (to grow the window), s to
+shrink it etc. as many times as needed. To exit the mode, press <escape>.
+
+.. note::
+    If using modes, users may also wish to use the Chord widget
+    (:class:`libqtile.widget.chord.Chord`) as this will display the name of the
+    currently active mode on the bar.
+
+Chains
+------
+
+Chords can also be chained to make even longer sequences.
+
+::
+
+    from libqtile.config import Key, KeyChord
+
+    keys = [
+        KeyChord([mod], "z", [
+            KeyChord([], "x", [
+                Key([], "c", lazy.spawn("xterm"))
+            ])
+        ])
+    ]
+
```

Comment:
Jesus lol, this is hilarious. I guess this is a side effect of how they are implemented, rather than this being specifically desired from the beginning? What use cases do people have for this?

Reply:
The question came up on the [qtile-dev group](https://groups.google.com/u/1/g/qtile-dev/c/DxbRCqtw-s8/m/nHTZWZXhCQAJ) so I included an example here.

I don't think it was envisaged but, as you say, it works because of the way the code is implemented.

Reply:
I like it.